content/doc/book/pipeline/development.adoc

```diff
@@ -0,0 +1,113 @@
+---
+layout: section
+title: Pipeline Development Tools
+---
+:description:
+:author:
+:email: jenkinsci-docs@googlegroups.com
+:sectanchors:
+:toc:
+:imagesdir: /doc/book/resources
+:hide-uri-scheme:
+
+= Pipeline Development Be
+
+In this section, we'll discuss tools and resources
+that help speed development of Jenkins Pipelines.
+
+== Creating
+
+The
+<<getting-started#built-in-documentation, built-in documentation>> and the
+<<getting-started#snippet-generator, Snippet Generator>>
+are good place to start when developing Jenkins Pipeline.
+
+=== Blue Ocean Editor (Declarative only)
+
+The
+<<../blueocean/pipeline-editor#, Blue Ocean Pipeline Editor>> provides a WYSIWYG
+way to create Delcarative Pipelines.
+
+=== Pipeline Unit Testing Framework
+
+The link:https://github.com/lesfurets/JenkinsPipelineUnit[Pipeline Unit Testing Framework]
+allows you to
+link:https://en.wikipedia.org/wiki/Unit_testing[unit test]
+Pipelines and <<shared-libraries, global shared libraries>>
+before running them like.   It provides a mock execution environment where real
+Pipeline steps are replaced with mock objects that you can use to check for expected
+behavior.
+
+New and rough around the edges, but promising.
+
+
+== Debugging
+
+Debugging failures in Pipelines can be difficult and time consuming.
+
+=== Replay Modified Pipline Runs
+
+The "Replay" feature lets you quick test small modifications to a Pipeline
+without creating a new commit or following regular code creation.
+
+[WARNING]
+====
+When using this feature for larger modifications save your changes
+to a file or editor outside of Jenkins before running them.
+Syntax errors may cause a pipeline run to not allow that run to be replayed,
+preventing you from replaying that failed run, meaning your changes will be lost.
+====
+
+=== Command-line Pipeline Linter (Declarative only)
+
+Jenkins can validate, or
```

Comment:
I think re-ordering as suggested above also means you can segue nicely. The Blue Ocean Pipeline Editor has some built-in validation support which is built on top of the Declarative Linter sub-system which is accessible from the CLI as well.